packages/amplify-storage-simulator/src/server/S3server.ts

```diff
@@ -0,0 +1,327 @@
+          "Size"         : statSync(files[file]).size,
+          "ETag"         : etag(files[file]),
+          "StorageClass" : 'STANDARD'
+        });
+        //console.log(request.params.path + files[file].split(dirPath)[1]);
+        keyCount = keyCount +1 ;
+      }
+    }
+    ListBucketResult["Name"] = this.route.split('/')[1];
+    ListBucketResult["Prefix"] = request.query.prefix || '';
+    ListBucketResult["KeyCount"] =  keyCount;
+    ListBucketResult["MaxKeys"]  = maxKeys;
+    ListBucketResult["Delimiter"] = delimiter;
+    if(keyCount === maxKeys){
+      ListBucketResult["IsTruncated"] =  true;
+    }
+    else{
+      ListBucketResult["IsTruncated"] =  false;
+    }
+    response.set('Content-Type', 'text/xml');
+    response.send(o2x({
+      '?xml version="1.0" encoding="utf-8"?': null,
+      ListBucketResult
+    }));
+  }
+
+  private async handleRequestDelete(request, response) {
+    // fill in  this content
+    // console.log("enter delete");
+    const filePath = join(this.localDirectoryPath, request.params.path);
+    if(existsSync(filePath)){
+      unlink(filePath, (err) => {
+        if (err) throw err;
+        response.send(xml(convert.json2xml(JSON.stringify(request.params.id + 'was deleted'))));
+      });
+    }
+    else{
+      response.sendStatus(204);
+    }
+  }
+
+  private async handleRequestPut(request, response) {
+    // fill in  this content
+    // console.log("put entered");
+    const directoryPath = join(String(this.localDirectoryPath), String(request.params.path));
+    ensureFileSync(directoryPath);
+    var new_data = stripChunkSignaturev2(request.body);
+    writeFileSync(directoryPath, new_data);
+    response.send(xml(convert.json2xml(JSON.stringify('upload success'))));
+  }
+  private async handleRequestPost(request, response) {
+    // fill in  this content
+    // console.log("post entered");
+    const directoryPath = join(String(this.localDirectoryPath), String(request.params.path));
+
+    if (request.query.uploads !== undefined) {
+      this.uploadId = uuid();
+      //response.set('Content-Type', 'text/xml');
+      response.send(o2x({
+        '?xml version="1.0" encoding="utf-8"?': null,
+        InitiateMultipartUploadResult: {
+          "Bucket": this.route,
+          "Key": request.params.path,
+          "UploadId": this.uploadId
+        }
+      }));
+    }
+    if (request.query.uploadId === this.uploadId) {
+      response.set('Content-Type', 'text/xml');
+      response.send(o2x({
+        '?xml version="1.0" encoding="utf-8"?': null,
+        CompleteMultipartUploadResult: {
+          "Location": request.url,
+          "Bucket": this.route,
+          "Key": request.params.path,
+          "Etag": etag(directoryPath) 
+        }
+      }));
+    }
+  }
+}
+
+// removing chunk siognature from request payload if present
+function stripChunkSignaturev2(buf: Buffer) {
+  let str = buf.toString();
+  var regex1 = /^[A-Fa-f0-9]+;chunk-signature=[0-9a-f]{64}/gm;
```

Comment:
why regex1

Reply:
changed the name to regex. Useful to know the position of signature in buffer